### DISCUSSION TITLE
Add free from "tags" field in the dataset + column model

### DISCUSSION THREAD

**Discussion by cccs-jc** [apache/superset#18467] [Ideas]
**Is your feature request related to a problem? Please describe.**
Superset can render dates using a data time pattern configured per column.

We would like to render other business types like IP, email etc. For example rendering a link to other web apps or formatting the data which is stored differently then how it is rendered. We are implementing a custom viz to do this. However we need a means to attach metadata information to the columns in the dataset model.

**Describe the solution you'd like**
We would like to be able to "tag" columns with additional information. Perhaps something similar to the "extra" field found on the dataset but on a column basis and also available on the client side.

**Describe alternatives you've considered**
At the moment we are using the type field of the column and setting it to IPv4, Domain, Email. However this field is probably not the best choice.

<img width="99" alt="column-metadata" src="https://user-images.githubusercontent.com/56140112/112218891-afa76480-8bfa-11eb-941f-fc671ad45e2a.png">

**Comment by junlincc** [comment]
@cccs-jc the request you have is similar to in-place adding column description on semantic layer? 
we do have plan to implement it, but would love additional help from you :)
